src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1210,23 +1210,43 @@ public static SqlConnectionStringBuilder CreateConnectionStringBuilder(Connectio
         /// </summary>
         /// <param name="ownerUri">URI of the owner of the connection</param>
         /// <param name="newDatabaseName">Name of the database to change the connection to</param>
-        public void ChangeConnectionDatabaseContext(string ownerUri, string newDatabaseName)
+        public void ChangeConnectionDatabaseContext(string ownerUri, string newDatabaseName, bool force = false)
         {
             ConnectionInfo info;
             if (TryFindConnection(ownerUri, out info))
             {
                 try
                 {
-                    foreach (DbConnection connection in info.AllConnections)
+                    info.ConnectionDetails.DatabaseName = newDatabaseName;
+
+                    foreach (KeyValuePair<string, DbConnection> pair in info.ConnectionMap)
                     {
-                        if (connection.State == ConnectionState.Open)
-                        {
-                            connection.ChangeDatabase(newDatabaseName);
-                        }
+                        if (pair.Value.State == ConnectionState.Open)
+                            {
+                                try
+                                {
+                                    pair.Value.ChangeDatabase(newDatabaseName);
+                                }
+                                catch (SqlException)
+                                {
+                                    if (force == true) {

```

Comment:
nit: `if (force)` also works here.
also, please put the `{` character on a separate line in C#